pkg/deploy/deploy.go

```diff
@@ -164,9 +165,18 @@ func deployGraph(ctx context.Context, configGraph *simple.DirectedGraph, clients
 
 		for _, root := range roots {
 			node := root.(graph.ConfigNode)
-			go func(ctx context.Context, node graph.ConfigNode) {
-				errChan <- deployNode(ctx, node, configGraph, clients, resolvedEntities)
-			}(context.WithValue(ctx, log.CtxKeyCoord{}, node.Config.Coordinate), node)
+			switch deployStrategy := api.NewAPIs()[node.Config.Coordinate.Type].DeployStrategy.(type) {
+			case api.SeqDeployStrategy:
```

Comment:
I like the idea in general, but limiting it to just config APIs seems limited and means you need to check if it even is an API here (does the switch actually work for other settings or blow up accessing the DeployStratgy if the config is not an API type?)

Reply:
go returns the zero value for the map's value type if the key is not found, so it should not blow up.

Reply:
my main point was more that the DeployStrategy feels like a general concept, but is limited to API types and needs you to check the API map in two places now. Maybe we could set it on the config level?